numcodecs/tests/common.py

```diff
@@ -276,7 +276,7 @@ def check_backwards_compatibility(codec_id, arrays, codecs, precision=None, pref
                 with open(codec_fn, mode='w') as cf:
                     _json.dump(codec.get_config(), cf, sort_keys=True, indent=4)
             # load config and compare with expectation
-            with open(codec_fn, mode='r') as cf:
+            with open(codec_fn) as cf:
```

Comment:
Could we keep the `mode` here? Explicit is better than implicit

Reply:
I understand, but [pyupgrade](https://github.com/asottile/pyupgrade) or [refurb](https://github.com/dosisod/refurb) disagree and believe implicit is better than explicit :smile:

I believe the rationale is that [open()](https://docs.python.org/3/library/functions.html#open) has many default arguments, and all of them are implicit (_buffering_, _encoding_, _errors_, _newline_, _closefd_, _opener_). That's why Python has default arguments. Why make an exception for _mode_?

Reply:
But then I don't have strong opinions on the subject. I was myself rather reluctant initially:
https://github.com/codespell-project/codespell/pull/2364#discussion_r997742025

Reply:
Yeah `mode` seems special since it affects whether the file may be modified (or even truncated) when opening

That said, not too worried about it given the placement (in the tests)

Have occasionally run into unexpected default behavior around `mode` in some libraries. Hence the initial pause